src/battle_ai_switch_items.c

```diff
@@ -27,6 +27,15 @@ static bool32 AiExpectsToFaintPlayer(void);
 static bool32 AI_ShouldHeal(u32 healAmount);
 static bool32 AI_OpponentCanFaintAiWithMod(u32 healAmount);
 
+static bool32 IsAceMon(u32 battlerId, u32 monPartyId)
+{
+    if (AI_THINKING_STRUCT->aiFlags & AI_FLAG_ACE_POKEMON
+        && !(gBattleStruct->forcedSwitch)
```

Comment:
Should this say `!(gBattleStruct->forcedSwitch & gBitTable[battlerId])`? (Maybe it shouldn't, I'm not too sure why there's a `forcedSwitch` bit for each battler.)

Reply:
Yeah, you're right, I fixed it